casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -17,24 +17,25 @@ object Estimator {
 
   implicit val decreasingOrder = Ordering[Int].reverse
 
-  def tips(childMap: HashMap[BlockHash, HashSet[BlockHash]],
-           blockLookup: HashMap[BlockHash, BlockMessage],
-           latestMessages: HashMap[Validator, BlockHash],
-           genesis: BlockMessage): IndexedSeq[BlockMessage] = {
+  def tips(blockDag: BlockDag, genesis: BlockMessage): IndexedSeq[BlockMessage] = {
     @tailrec
     def sortChildren(blocks: IndexedSeq[BlockHash],
+                     childMap: HashMap[BlockHash, HashSet[BlockHash]],
                      scores: collection.Map[BlockHash, Int]): IndexedSeq[BlockHash] = {
       // TODO: This ListContrib.sortBy will be improved on Thursday with Pawels help
       val newBlocks =
         ListContrib
-          .sortBy[BlockHash, Int](blocks.flatMap(replaceBlockHashWithChildren).distinct, scores)
+          .sortBy[BlockHash, Int](
+            blocks.flatMap(replaceBlockHashWithChildren(childMap, _)).distinct,
+            scores)
       if (stillSame(blocks, newBlocks)) {
         blocks
       } else {
-        sortChildren(newBlocks, scores)
+        sortChildren(newBlocks, childMap, scores)
       }
     }
-    def replaceBlockHashWithChildren(b: BlockHash) = {
+    def replaceBlockHashWithChildren(childMap: HashMap[BlockHash, HashSet[BlockHash]],
+                                     b: BlockHash) = {
       val empty                 = new HashSet[BlockHash]()
```

Comment:
I know not part of this PR, but it's something that I must have missed before. This should be `HashSet.empty[BlockHash]`. There is only one empty immutable HashSet, (and HashMap too), so we should not instantiate it with `new`.